e2e/e2e.py

```diff
@@ -17,327 +22,587 @@
 MOBILE = HEADLESS
 USER_RESPONSE_TIME=0.2 # A typical user cannot click faster than once every 200ms
 
-def fill_field(driver, selector, content):
-    try:
-        wait = WebDriverWait(driver, 10) # wait up to 10s to find the element
-        input = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, selector)))
-        input.clear()
-        input.send_keys(content)
-    except Exception as e:
-        print("Failed when filling in field '{}'".format(selector))
-        raise e
-
-def click_button(driver, selector):
-    try:
-        print("Clicking button: {}".format(selector))
-        # Force a small wait to mimick a real user. This is to stabilize these tests a bit more
-        time.sleep(USER_RESPONSE_TIME)
-        wait = WebDriverWait(driver, 10) # wait up to 10s to find the element
-        button = wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, selector)))
-        button.click()
-    except Exception as e:
-        print("Failed when clicking on button '{}'".format(selector))
-        raise e
-
-def click_menu_button(driver, hover_selector, button_selector):
-    try:
+class PlayerDriver():
+    def __init__(self):
+        options = Options()
+        if HEADLESS:
+            options.add_argument('-headless')
+        self.driver = webdriver.Chrome(options=options)
+
+    def close(self):
+        '''Close the browser'''
+        self.driver.close()
+
+    def ensure_no_errors(self):
+        '''Ensures that no error has been logged in the browser's console'''
+        logs = self.driver.get_log('browser')
+
+        errors = False
+        for log in logs:
+            print('[browser]' + textwrap.fill(log['message'], 120))
+            if log['level'] == 'SEVERE':
+                errors = True
+        if errors:
+            raise Exception('Errors encountered, stopping here.')
+
+    def go_to_page(self, url):
+        '''Visit an URL'''
+        self.driver.get(url)
+        time.sleep(1)
```

Comment:
WHY THE SLEEP ?

Reply:
pour être sûr que la page soit chargée, je commente pour clarifier

Reply:
y a pa de wait-for fonctionnel là ?

Reply:
semblerait qu'en attendant la présence de app-root on est bon !